### PR TITLE
*: refactor, format, and update

### PR DESCRIPTION
**Pull request by VoltrexKeyva**
- Rewrite to ESM.
- Refactor the JavaScript files.
- Update dependencies and developer dependencies to their latest versions.
- Format files with Prettier.
- Add `package-lock.json` to `.gitignore`.
- Add missing backtick in `README.md`.

### CODE REVIEW

**Comment by VoltrexKeyva** [comment]
Tagging @rvagg and @Trott since this repository isn't active.

**Review by rvagg — CHANGES_REQUESTED** [review]
seems fine to me, but involving my personal code style preference I wouldn't mind seeing:

* Apply `standard` to this, and in particular ditch the semicolons, they're poking my eyes
* Single-line `if`s are gross (don't look at my old code pls) and hard to parse, these days I'd much prefer explicit bracing of every `if` and `else`
* Some random blank lines that don't seem to have a pattern, e.g. line 42. I like spacing out code, as long as there's a rhyme to is. Otherwise random blank lines are weird.

Also, thanks for giving it some love!

File: commit-stream.js [hunk 1]
```diff
@@ -1,95 +1,116 @@
...
+    } else if (
+      (m = line.match(/^\s+Reviewed[- ]?By:?\s*([^<]+) <([^>]+)>\s*$/i)) !==
+      null
+    ) {
+      if (commit.reviewers === undefined || commit.reviewers === null) {
         commit.reviewers = []
+      }
       commit.reviewers.push({ name: m[1], email: m[2] })
-    } else if (m = line.match(/^\s+PR(?:[- ]?URL)?:?\s*(.+)\s*$/)) {
+    } else if ((m = line.match(/^\s+PR(?:[- ]?URL)?:?\s*(.+)\s*$/)) !== null) {
```
Comment (rvagg) [inline commit-stream.js:49]: interesting, did standard complain about this or are you just opting to be explicit (not objecting, just intested)

**Review by rvagg — APPROVED** [review]
lgtm, thanks for indulging my preferences!

File: commit-stream.js (hunk 1 above)
Comment (VoltrexKeyva) [reply commit-stream.js:49]: The standard tool didn't complain about it, I'm just making it explicit.

**Comment by rvagg** [comment]
See #8, I've pulled this commit in to there and would like to merge over there than here so we can get an auto-publish flow working.

**Comment by rvagg** [comment]
included in v2.0.0, thanks!

https://www.npmjs.com/package/commit-stream/v/2.0.0
https://github.com/nodejs/commit-stream/releases/tag/v2.0.0
